File: api/src/main/java/club/yunzhi/smartcommunity/entity/Cult.java
package club.yunzhi.smartcommunity.entity;

import org.hibernate.annotations.SQLDelete;
;

import javax.persistence.Column;
import javax.persistence.Entity;
import java.sql.Timestamp;

/**
 * 邪教
 */
@Entity
@SQLDelete(sql = "update `cult` set deleted = 1 where id = ?")
public class Cult extends SoftDeleteEntity {
  public static final String TABLE_NAME = "cult";
  @Column(nullable = false)
  private String name = "";

  /**最后一次的使用时间*/
  private Timestamp lastUsedTime;

  public Timestamp getLastUsedTime() {
    return lastUsedTime;
  }

  public void setLastUsedTime(Timestamp lastUsedTime) {
    this.lastUsedTime = lastUsedTime;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }
}

File: web/projects/lib/src/entity/cult.ts
/**
 * 邪教
 */
export class Cult {
  id: number;
  /**名字（唯一）*/
  name: string;
  /**最后一次的使用时间*/
  lastUsedTime: number;
  constructor(data = {} as { id: number, name: string, lastUsedTime?: number}) {
    if (data) {
      this.id = data.id;
      this.name = data.name;
      this.lastUsedTime = data.lastUsedTime;
    }
  }
}
